pilot/pkg/xds/endpoints/endpoint_builder.go

```diff
@@ -661,7 +661,8 @@ func buildEnvoyLbEndpoint(b *EndpointBuilder, e *model.IstioEndpoint, mtlsEnable
 	tunnel := supportTunnel(b, e)
 	// Only send HBONE if its necessary. If they support legacy mTLS and do not explicitly PreferHBONE, we will use legacy mTLS.
 	// However, waypoints (TrafficDirectionInboundVIP) do not support legacy mTLS, so do not allow opting out of that case.
-	if mtlsEnabled && !features.PreferHBONESend && b.dir != model.TrafficDirectionInboundVIP {
+	supportsMtls := e.TLSMode == model.IstioMutualTLSModeLabel
+	if supportsMtls && !features.PreferHBONESend && b.dir != model.TrafficDirectionInboundVIP {
```

Comment:
Will this exclude workload waypoints?

Reply:
I am not sure I follow the relation between this change and workload waypoints?

Reply:
The inbound vip part; I thought that was only a thing for service waypoints

Reply:
ah , that is existing code. I guess that might mean we end up preferring mTLS (instead of hbone) in that case. Which should work fine, either *should* work it's more about what is preferred. 

Haven't used workload waypoint much so not really sure

Reply:
Ack, good enough for me for now